modules/metrics/src/main/java/org/apache/ignite/internal/metrics/exporters/PushMetricExporter.java

```diff
@@ -62,7 +57,7 @@ public void start() {
 
                 throw th;
             }
-        }, period, period, TimeUnit.MILLISECONDS);
+        }, period(), period(), TimeUnit.MILLISECONDS);
```

Comment:
1. `new NamedThreadFactory("metrics-exporter", log)` - it seems to me, it would be nice to add a name of this exporter as well. Let's assume that a user configure two different push exporters.
2. should `fut` and `schedular` fields be volatile?
3. is it possible, that `period()` returns a negative value? in that case `scheduleWithFixedDelay` throws `IllegalArgumentException` and so `PushMetricExporter#stop()` will throw `NullPointerException`